server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/HistogramMultiBucketsAggregatorReducer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.search.aggregations.bucket.histogram;
+
+import org.elasticsearch.common.util.LongObjectPagedHashMap;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.search.aggregations.AggregationReduceContext;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.aggregations.InternalMultiBucketAggregation;
+import org.elasticsearch.search.aggregations.bucket.MultiBucketAggregatorsReducer;
+import org.elasticsearch.search.aggregations.bucket.MultiBucketsAggregation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  Reduces aggregations where buckets are represented by a o long key. It uses a {@link LongObjectPagedHashMap}
+ *  to keep track of the different buckets.
+ */
+abstract class HistogramMultiBucketsAggregatorReducer<B extends MultiBucketsAggregation.Bucket> implements Releasable {
```

Comment:
I think this only works with long-keyed buckets. But that's fine. Maybe a rename?